src/payload.rs

```diff
@@ -481,7 +480,13 @@ impl Iterator for ZBytesInIterator {
         let mut data = MaybeUninit::<z_owned_bytes_t>::uninit();
 
         (self.body)(&mut data, self.context);
```

Comment:
I think we need to change body to return false when iteration ends, instead of treating empty z_owned_bytes_t as end